File: 알고리즘/선택정렬/selectionSort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n):
        min_idx = i
        for j in range(i+1,n):
            if arr[min_idx] > arr[j]:
                min_idx = j
        arr[min_idx], arr[i] = arr[i], arr[min_idx]
    return arr

arr = [64, 25, 12, 22, 11]
print(selection_sort(arr))      

File: 알고리즘/선택정렬/selectionSort.kt
fun selectionSort(arr : IntArray) : IntArray{
    val n = arr.size
    for(i in 0 until n-1){
        var minIndex = i
        for(j in i+1 until n){
            if(arr[minIndex] > arr[j]){
                minIndex = j
            }
        }
        val temp = arr[minIdex]
        arr[minIndex] = arr[i]
        arr[i] = temp
    }
    return arr
}

fun main() {
    val arr = intArrayOf(64, 25, 12, 22, 11)
    selectionSort(arr)
    println(arr.joinToString(" "))
}
